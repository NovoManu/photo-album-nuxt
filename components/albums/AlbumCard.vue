<template>
  <div class="card d-flex">
    <v-card
      class="card-album"
      :style="className"
      ripple
      tile
      @click="redirectToAlbum"
    >
      <v-img
        v-observe-visibility="visibilityChanged"
        :src="`https://via.placeholder.com/150/${colors[album.userId]}/808080`"
      />
      <div v-if="visible" class="description">
        <span class="my-caption text-xs-center">{{ album.title }}</span>
        <small>by</small>
        <span>
          {{ user.name || 'User not found' }}
        </span>
      </div>
    </v-card>
  </div>
</template>
<script lang="ts">
import { Component, Vue, Prop, namespace } from 'nuxt-property-decorator'
import { Album, User } from '~/utils/types'
import colors from '~/utils/colors.ts'

const UsersModule = namespace('users')

@Component
export default class AlbumCard extends Vue {
  @Prop({ required: true }) album!: Album
  @UsersModule.Getter getUserById!: any
  colors: string[] = colors
  visible: boolean = false
  get user(): User | {} {
    return this.getUserById(this.album.userId) || {}
  }
  get className() {
    return this.visible ? {} : { border: 'none' }
  }
  redirectToAlbum() {
    this.$nuxt.$router.push(`/${this.album.id}/photos`)
  }
  visibilityChanged(isVisible) {
    this.visible = isVisible
  }
}
</script>
