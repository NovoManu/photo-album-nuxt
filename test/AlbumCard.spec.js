import { mount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import AlbumCard from '@/components/albums/AlbumCard.vue'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('index.vue', () => {
  let actions
  let store
  beforeEach(() => {
    actions = {
      actionClick: jest.fn(),
      actionInput: jest.fn()
    }
    store = new Vuex.Store({
      actions
    })
  })
  const wrapper = mount(AlbumCard, {
    propsData: {
      album: {}
    },
    store,
    localVue
  })
  it('renders component', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
  it('has initial colors', () => {
    expect(wrapper.vm.colors).toBeDefined()
  })
  it('has initial visible', () => {
    expect(wrapper.vm.visible).toBe(false)
  })
  it('getUserById not called by click', () => {
    expect(actions.actionClick).not.toHaveBeenCalled()
  })
})
