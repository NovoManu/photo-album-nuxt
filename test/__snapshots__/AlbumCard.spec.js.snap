// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index.vue renders component 1`] = `
<div
  class="card d-flex"
>
  <v-card
    class="card-album"
    ripple=""
    tile=""
  >
    <v-img
      src="https://via.placeholder.com/150/undefined/808080"
    />
     
    <!---->
  </v-card>
</div>
`;
